ukat/mapping/mtr.py

```diff
@@ -0,0 +1,106 @@
+import os
+import numpy as np
+import nibabel as nib
+
+
+class MTR:
+    """
+    Generates a Magnetisation Transfer Ratio (MTR) map from a series of volumes
+    collected with 2 different magnetisation transfer values (ON and OFF).
+
+    Attributes
+    ----------
+    mtr_map : np.ndarray
+        The estimated magnetisation transfer ratio values.
+        Each value should be a float between 0 and 1.
+    shape : tuple
+        The shape of the MTR map.
+    mt_on : np.ndarray
+        The array corresponding to the magnetisation transfer value ON.
+    mt_off : np.ndarray
+        The array corresponding to the magnetisation transfer value OFF.
+    mask : np.ndarray
+        A boolean mask of the voxels to fit.
+    """
+
+    def __init__(self, pixel_array, affine, mask=None):
+        """Initialise a MTR class instance.
+
+        Parameters
+        ----------
+        pixel_array : np.ndarray
+            A 4D/3D array containing the image at each magnetisation transfer
+            value i.e. the dimensions of the array are
+            [x, y, 2] or [x, y, z, 2].
```

Comment:
It doesn't say anywhere which order on/off should be in i.e. should `[..., 0]` be on or off.

Reply:
I have a comment in the code with a similar explanation so I should bring it to this docstring then?

Reply:
Yeah, docstrings appear as tool tips/help in IDEs e.g. in Spyder (the IDE I use when I'm doing data analysis) ctrl + I opens the docstring of the function you're clicked on so you know what parameters it takes etc. Also some documentation packages make their documentation websites from the docstring. Are a rule of thumb, comments in the code should be about the mechanics of how code operates to make it easier for developers in the future, docstrings should contain all the information needed for a user.